controllers/cinderapi_controller.go

```diff
@@ -252,6 +259,54 @@ func (r *CinderAPIReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Man
 		return nil
 	}
 
+	// index passwordSecretField
+	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &cinderv1beta1.CinderAPI{}, passwordSecretField, func(rawObj client.Object) []string {
+		// Extract the secret name from the spec, if one is provided
+		cr := rawObj.(*cinderv1beta1.CinderAPI)
+		if cr.Spec.Secret == "" {
+			return nil
+		}
+		return []string{cr.Spec.Secret}
```

Comment:
The purpose of these blocks (we'll look in a follow up if we can make this more generic) is to add the named secret (which the we don't own), if filled/provided, to the index so that it can be watched for changes and the controller reconciles, e.g. pw secret change, cert secret change, ... . More details can be found in [Watching Externally Managed Resources](https://kubebuilder.io/reference/watching-resources/externally-managed#watching-externally-managed-resources)